### PR TITLE
added cornerRadius for hover style

### PR DESCRIPTION
**Pull request by gokul1099**
This PR is an experimental fix for #12 . In this PR I have added method to apply the borderRadius style which is applied to View to hoverStyle. Currently I have added support for borderRadius style. But for other cases like borderTopEndRadius, borderTopStartRadius will not work.
## Changelog:
[VISIONOS] - added method to apply borderRadius to hoverStyle

### CODE REVIEW

File: packages/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.mm [hunk 1]
```diff
@@ -287,14 +287,20 @@ - (void)updateProps:(const Props::Shared &)props oldProps:(const Props::Shared &
...
-  if (oldViewProps.borderStyles != newViewProps.borderStyles || oldViewProps.borderRadii != newViewProps.borderRadii ||
+  if (oldViewProps.borderStyles != newViewProps.borderStyles ||
       oldViewProps.borderColors != newViewProps.borderColors) {
     needsInvalidateLayer = YES;
   }
-    
+// 'borderRadii'
+    if (oldViewProps.borderRadii != newViewProps.borderRadii) {
+        needsInvalidateLayer = YES;
+        CGFloat borderRadius = newViewProps.borderRadii.all ?newViewProps.borderRadii.all.value() : 0.0;
```
Comment (okwasniewski) [inline packages/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.mm:297]: What about the case where we have different border radious on each corner? I think this should also be handled here

**Review by okwasniewski — COMMENTED** [review]
In general the solution looks good, there is something off with formatting if you could check it out (you can use CTRL + I to format).

There is a need to handle additional case there

File: packages/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.mm (hunk 1 above)
Comment (gokul1099) [reply packages/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.mm:297]: yes, currently it supports only borderRadius. just want to confirm the proposed solution is correct. I will work on the other cases.

**Comment by gokul1099** [comment]
@okwasniewski  in updateHoverEffect we are using rectShapeWithCornerRadius for creating a rect shape for Hover effect. Here we can pass only one corner radius which will work in case of borderRadius props. if we pass separate cornerRadius for each corner this will not work as we can use only one CornerRadius in rectShapeWithCornerRadius api. Can we use UIBeizerPath for this. So that we can contruct a separate path for each cornerRadius and append all the path and forms a rect shape with multiple cornerRadius and use that shape for hover style.

**Comment by gokul1099** [comment]
@okwasniewski in updateHoverEffect we are using rectShapeWithCornerRadius for creating a rect shape for Hover effect. Here we can pass only one corner radius which will work in case of borderRadius props. if we pass separate cornerRadius for each corner this will not work as we can use only one CornerRadius in rectShapeWithCornerRadius api. Can we use UIBeizerPath for this. So that we can contruct a separate path for each cornerRadius and append all the path and forms a rect shape with multiple cornerRadius and use that shape for hover style.

**Comment by okwasniewski** [comment]
> @okwasniewski in updateHoverEffect we are using rectShapeWithCornerRadius for creating a rect shape for Hover effect. Here we can pass only one corner radius which will work in case of borderRadius props. if we pass separate cornerRadius for each corner this will not work as we can use only one CornerRadius in rectShapeWithCornerRadius api. Can we use UIBeizerPath for this. So that we can contruct a separate path for each cornerRadius and append all the path and forms a rect shape with multiple cornerRadius and use that shape for hover style.

Sounds reasonable. I will merge this PR and create a new ticket to handle this case. If you are willing to continue working on this, I can assign you 👍🏻

**Pull request merged**